internal/querynode/shard_cluster_service.go

```diff
@@ -101,6 +101,20 @@ func (s *ShardClusterService) releaseShardCluster(vchannelName string) error {
 	return nil
 }
 
+func (s *ShardClusterService) close() error {
+	log.Debug("close shard cluster service")
+	s.clusters.Range(func(key, value any) bool {
+		cs, ok := value.(*ShardCluster)
+		if !ok {
+			log.Error("convert to ShardCluster fail", zap.Any("key", key))
```

Comment:
If type assertion failed, we should return here to avoid calling `close` on `nil`.